sc2/game_state.py

```diff
@@ -81,50 +81,69 @@ def positions(self) -> List[Point2]:
+        self.action_errors = response_observation.action_errors  # error actions since last loop
         # https://github.com/Blizzard/s2client-proto/blob/51662231c0965eba47d5183ed0a6336d5ae6b640/s2clientprotocol/sc2api.proto#L575
         # TODO: implement alerts https://github.com/Blizzard/s2client-proto/blob/51662231c0965eba47d5183ed0a6336d5ae6b640/s2clientprotocol/sc2api.proto#L640
         self.observation = response_observation.observation
         self.player_result = response_observation.player_result
         self.chat = response_observation.chat
         self.common: Common = Common(self.observation.player_common)
-        self.psionic_matrix: PsionicMatrix = PsionicMatrix.from_proto(self.observation.raw_data.player.power_sources) # what area pylon covers
-        self.game_loop: int = self.observation.game_loop # game loop, 22.4 per second on faster game speed
+        self.psionic_matrix: PsionicMatrix = PsionicMatrix.from_proto(
+            self.observation.raw_data.player.power_sources
+        )  # what area pylon covers
+        self.game_loop: int = self.observation.game_loop  # game loop, 22.4 per second on faster game speed
 
-        self.score: ScoreDetails = ScoreDetails(self.observation.score) # https://github.com/Blizzard/s2client-proto/blob/33f0ecf615aa06ca845ffe4739ef3133f37265a9/s2clientprotocol/score.proto#L31
-        self.abilities = self.observation.abilities # abilities of selected units
-        destructables = [x for x in self.observation.raw_data.units if x.alliance == 3 and x.radius > 1.5] # all destructable rocks except the one below the main base ramps
-        self.destructables: Units = Units.from_proto(destructables, game_data)
+        self.score: ScoreDetails = ScoreDetails(
+            self.observation.score
+        )  # https://github.com/Blizzard/s2client-proto/blob/33f0ecf615aa06ca845ffe4739ef3133f37265a9/s2clientprotocol/score.proto#L31
+        self.abilities = self.observation.abilities  # abilities of selected units
 
         # Fix for enemy units detected by my sensor tower, as blips have less unit information than normal visible units
-        visibleUnits, hiddenUnits = [], []
+        visibleUnits, hiddenUnits, minerals, geysers, = [], [], [], []
+        destructables, enemy, own = [], [], []
         for u in self.observation.raw_data.units:
-            hiddenUnits.append(u) if u.is_blip else visibleUnits.append(u)
+            if u.is_blip:
+                hiddenUnits.append(u)
+            else:
+                visibleUnits.append(u)
+                # all destructable rocks except the one below the main base ramps
+                if u.alliance == 3 and u.radius > 1.5:
+                    destructables.append(u)
+                if u.alliance == 3:
+                    # mineral field enums
+                    if u.unit_type in {146, 147, 341, 483, 665, 666, 796, 797, 884, 885, 886, 887}:
```

Comment:
I would exchange the numbers for the actual names(f.e -RICHMINERALFIELD ), since these numbers are "magic numbers", also the numbers are always changing every protocol

Reply:
yeah you are right, i will change that